### DISCUSSION TITLE
Hash code for an inflight closure

### DISCUSSION THREAD

**Discussion by eladb** [winglang/wing#6184] [General]
We need a way to identify when the generated code for an inflight closure (and all of its deps) have changed in order to minimize updates when loading new code into the simulator.

For cloud targets, we use the `esbuild` bundle hash during pre-synth (after captures are resolved), but since we are not doing preflight bundling when compiling to the simulator, we don't have this information.

A deterministic hash of the inflight closure can go a long way to simplify downstream caching, etc.

Any ideas?

**Comment by ekeren** [comment]
We remove esbuild for sim because of performance reasons, but since then I believe we moved away from using wasm esbuild which was slow. Maybe we can come back to using esbuild for sim again?

If we can use esbuild, we might be able to get the list of dependencies path (https://github.com/evanw/esbuild/issues/673). 


@MarkMcCulloh?

**Reply by MarkMcCulloh** [reply]
While using the native esbuild is much faster, it's still pretty slow compared to how fast sim compilation is. I do think it would be worth trying again though.

**Reply by ekeren** [reply]
@MarkMcCulloh , can you estimate how long will it take to check if this solution is valid?
